packages/core/src/components/types.ts

```diff
@@ -41,3 +44,10 @@ type OmitCommonProps<
   Target,
   OmitAdditionalProps extends keyof any = never
 > = Omit<Target, "transition" | "as" | "color" | OmitAdditionalProps>;
+
+export type StyleProps = StyledProps &
```

Comment:
I reckon `ComponentProps` would be a better name than `StyleProps`, as this includes props that aren't related to styling.